Solidigm D7-P5520 Firmware upgrade - Guide creation + FR translation

SlimJ4 opened the pull request:
<!--
Hello 👋
Thank you for submitting a Pull Request.

For Work In Progress Pull Requests, please use the Draft PR feature, see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details. As long as your Pull Request is a draft, the OVHcloud Guides Team will not start proofreading it.

For a timely review/response, please avoid force-pushing additional commits if your PR already received reviews or comments.

DO NOT USE a fork if you are an OVHcloud employee. Instead, please contact the OVHcloud Guides Team so to be granted /write access to this repository.

Feel free to apply labels on your Pull Request, selecting them from the list to your right.

Before submitting a Pull Request, please ensure you have:

- 📖 Read the OVHcloud Documentation readme: https://github.com/ovh/docs/blob/develop/readme.md
- 🇬🇧 Edited (at least) the en-gb version of a guide.
- 🇫🇷 Edited also the fr-fr version if you are a French speaker.
- 🌐 Edited all the files if you're doing a fix or a minor update (this is not mandatory but recommended).
- 👷‍♀️ Created a small PR, if applicable.
- ✅ Tested every step you're documenting.
- 📝 Used descriptive commit messages.
- 📐 Resized images which width exceeds 1400px.
- 📝 Edited or blurred any sensitive/private information from your text and images (IPs, user IDs, private URLs, etc.).
-->

## What type of Pull Request is this?

<!-- Delete the lines which don't apply: -->
- New guide(s) <!-- You need to create a meta.yaml file and edit the index: https://github.com/ovh/docs/blob/develop/pages/index.md -->

## Description

<!-- Please provide a short description for your Pull Request, including (if applicable) a ticket reference (no internal URL!) or GitHub issue -->
Created a guide that helps users to update the firmware of Solidigm D7-P5520 SSDs on their Bare Metal servers, using information provided by our technical teams. French and English localisations available.

## Mandatory information

<!-- Delete the line which doesn't apply: -->
- This Pull Request can be merged as soon as possible.

SlimJ4 committed:
Guide creation + FR translation

Created the guide with provided instructions. FR translation done.

Y0Coss commented on pages/bare_metal_cloud/dedicated_servers/solidigm_d7-p5520_fw_update/en-gb.md:
@SlimJ4 , please name the files as follows:
- `guide.en-gb.md`
- `guide.fr-fr.md`

SlimJ4 replied:
@Y0Coss done, somehow missed that before the PR

SlimJ4 committed:
File rename

SlimJ4 committed:
Word replacement

Replaced all instances of "microprogramme" with "firmware"

SlimJ4 committed:
Minor edits

Small cosmetic or language edits for consistency.

SlimJ4 committed:
Minor change

Edited for consistency.

SlimJ4 committed:
Collapsible sections + date

Added collapsible sections to make the article more streamlined.
Updated the date.

Y0Coss committed:
Proofreading

Y0Coss commented:
Thank you for your contribution @SlimJ4
